### ISSUE TITLE
Severe JobRunr Exception

### ISSUE THREAD

**Issue by zhemaituk** [jobrunr/jobrunr#138]
At some point jobs stopped execution. Enqueue worked fine, so jobs were just piling up. Possibly it was a thread starvation in the system for some time, which was resolved, but jobrunr didn't recover, had to restart the container.

### Runtime information
- __Timestamp__: 2021-05-29T22:32:46.419388Z
- __Location__: BackgroundJobServer 6dbfb758-d288-4b9e-8f4e-e25d9a02ddce
- __JobRunr Version__: Unable to determine version
- __StorageProvider__: org.jobrunr.storage.sql.postgres.PostgresStorageProvider
- __Java Version__: 15.0.2
- __Is running from nested jar__: false

### Background Job Servers
- BackgroundJobServer id: 6dbfb758-d288-4b9e-8f4e-e25d9a02ddce
(workerPoolSize: 4, pollIntervalInSeconds: 5, firstHeartbeat: 2021-05-29T22:18:06.406440Z, lastHeartbeat: 2021-05-29T22:32:46.417109Z)

#### Concurrent modified jobs:
Job id: 0afad781-369d-4e56-88fd-3e035ec94672
	Local version: 4; Storage version: 4
	Local state: PROCESSING; Storage state: SCHEDULED
	Local last updated: 2021-05-29T22:32:46.397074Z; Storage last updated: 2021-05-29T22:32:46.380517Z

### Exception
```java
org.jobrunr.SevereJobRunrException: Could not resolve ConcurrentJobModificationException
	at org.jobrunr.server.JobZooKeeper.processJobList(JobZooKeeper.java:199)
	at org.jobrunr.server.JobZooKeeper.updateJobsThatAreBeingProcessed(JobZooKeeper.java:147)
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:75)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset$$$capture(FutureTask.java:305)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
	at java.base/java.lang.Thread.run(Thread.java:832)

```

**Comment by rdehuyss** [comment]
Can you please provide me with the JobRunr version?

**Comment by zhemaituk** [comment]
2.0.0

**Comment by rdehuyss** [comment]
Can you please upgrade to the latest version, delete the exception by dismissing it and reopen the issue if it comes back?

A fix related to it was done in release 3.0.0
